File: app/models/disk_file.go
package models

import "time"

const (
	FILETYPE_FOLDER = "D"
	FILETYPE_FILE   = "F"
)

type DiskFile struct {
	ID         string    `json:"id" form:"id"`
	Name       string    `json:"name" form:"name"`
	FilePath   string    `json:"filePath" form:"filePath"`
	ParentPath string    `json:"parentPath" form:"parentPath"`
	FileType   string    `json:"fileType" form:"fileType"`
	UpdatedAt  time.Time `json:"updatedAt" form:"updatedAt"`
}

type DiskFileSearchCondition struct {
	ParentPath string `form:"parentPath" binding:"omitempty"`
	FilePath   string `form:"filePath" binding:"omitempty"`
	FileType   string `form:"fileType" binding:"omitempty"`
}

File: web/src/api/disk_file.ts
import { withPath, withQuery } from "fetch-utils";
import { fetchMany } from "./base";

export interface DiskFileAPI {
  searchDiskFiles(
    condition: DiskFileSearchCondition
  ): Promise<[number, DiskFile[]]>;
}

export type DiskFile = {
  id: string;
  name: string;
  filePath: string;
  parentPath: string;
  fileType: string;
  updatedAt: string;
};

export type DiskFileSearchCondition = {
  parentPath?: string;
  filePath?: string;
  fileType?: string;
};

export async function searchDiskFiles(
  condition: DiskFileSearchCondition
): Promise<[number, DiskFile[]]> {
  return await fetchMany(
    withPath("app/disk-files", "merge"),
    withQuery(condition)
  );
}
